Examples/source/InstrumentAbstraction/ForceVoltagePPMU.cs

```diff
@@ -17,75 +23,128 @@ namespace NationalInstruments.SemiconductorTestLibrary.Examples.InstrumentAbstra
         {
-            var pinNames = new string[] { "PinA", "PinB" };
             var sessionManager = new TSMSessionManager(tsmContext);
-            var ppmuPins = sessionManager.Digital(pinNames);
+            var ppmuPinNames = new[] { "PinA", "PinB" };
+            var perPinVoltageLevels = new[] { 1.0, 3.3 };
+            var activeSites = tsmContext.SiteNumbers.ToArray();
+            // Create a PinSiteData object to map each voltage level with the appropriate pin name.
+            var perPinLevelsToForce = new PinSiteData<double>(ppmuPinNames, activeSites, perPinVoltageLevels);
+
+            var ppmuPins = sessionManager.Digital(ppmuPinNames);
+
+            ppmuPins.ForceVoltage(perPinLevelsToForce);
+        }
 
-            var pinASettings = new PPMUSettings { VoltageLevel = 1.0, CurrentLimitRange = 0.09 };
-            var pinBSettings = new PPMUSettings { VoltageLevel = 3.3, CurrentLimitRange = 0.01 };
-            var smuSettings = new Dictionary<string, PPMUSettings>()
+        /// <summary>
+        /// This example demonstrates how to force different voltage levels and current limits for different pins across all sites.
+        /// The example assumes there are exactly 2 pins.
+        /// </summary>
+        /// <param name="tsmContext">The <see cref="ISemiconductorModuleContext"/> object.</param>
+        internal static void DifferentValuesPerPpmuPin(ISemiconductorModuleContext tsmContext)
+        {
+            var sessionManager = new TSMSessionManager(tsmContext);
+            // This dictionary is hard coded for demonstration purposes,
+            // but could otherwise be built by looping over array input parameters of the same length.
+            var perPinSmuSettings = new Dictionary<string, PPMUSettings>()
             {
-                [pinNames[0]] = pinASettings,
-                [pinNames[1]] = pinBSettings,
+                ["PinA"] = new PPMUSettings { VoltageLevel = 1.0, CurrentLimitRange = 0.09 },
+                ["PinB"] = new PPMUSettings { VoltageLevel = 3.3, CurrentLimitRange = 0.01 },
             };
 
-            ppmuPins.ForceVoltage(smuSettings);
+            var ppmuPins = sessionManager.Digital(perPinSmuSettings.Keys.ToArray());
+
+            ppmuPins.ForceVoltage(perPinSmuSettings);
         }
 
         /// <summary>
         /// This example demonstrates how to configure different voltage levels for each site.
-        /// The example assumes that there are 2 pins and 4 sites.
+        /// The example assumes there is a maximum of 4 sites.
         /// </summary>
         /// <param name="tsmContext">The <see cref="ISemiconductorModuleContext"/> object.</param>
-        internal static void DifferentValuesPerSitePpmu(ISemiconductorModuleContext tsmContext)
+        /// <param name="ppmuPinNames">The PPMU pins to force voltage on</param>
+        internal static void DifferentValuesPerSitePmmu(ISemiconductorModuleContext tsmContext, string[] ppmuPinNames)
```

Comment:
DifferentValuesPerSiteP**p**mu. And maybe also add Pins in the end, so DifferentValuesPerSitePpmuPins?

Reply:
@lornazh, I will go with `DifferentValuesPerSiteAcrossAllPpmuPins`.